js/deer-record.js

```diff
@@ -336,45 +337,48 @@ export default class DeerReport {
     }
 
     simpleUpsert(event) {
-        let record = {
-            "@type": this.type
-        }
-        if (this.context) { record["@context"] = this.context }
-        if (this.evidence) { record.evidence = this.evidence }
+        let record = {}
         Array.from(this.elem.querySelectorAll(DEER.INPUTS.map(s => s + "[" + DEER.KEY + "]").join(","))).map(input => {
             let key = input.getAttribute(DEER.KEY)
             let val = input.value
             let title = input.getAttribute("title")
             let evidence = input.getAttribute(DEER.EVIDENCE)
-
             if (title || evidence) {
                 val = { "@value": val }
                 if (title) val.name = title
                 if (evidence) val.evidence = evidence
             }
-
             record[key] = (record.hasOwnProperty(key)) ?
-                ((Array.isArray(record[key])) ? record[key].push(val) : [record[key], val]) :
-                val
-
-            let formId = this.elem.getAttribute(DEER.ID)
-            let action = "CREATE"
-
-            if (formId) {
-                action = "OVERWRITE"
-                record["@id"] = formId
-            }
-
-            return fetch(DEER.URLS[action], {
-                    method: (formId) ? "PUT" : "POST",
-                    headers: {
-                        "Content-Type": "application/json; charset=utf-8"
-                    },
-                    body: JSON.stringify(record)
-                })
-                .then(response => response.json())
-                .then(obj => input.setAttribute(DEER.ID, obj.new_obj_state["@id"]))
+                ((Array.isArray(record[key])) ? record[key].push(val) : [record[key], val]) : val
+        })
+        let formId = this.elem.getAttribute(DEER.ID)
+        let action = "CREATE"
+        if (formId) {
+            action = "OVERWRITE"
+            record["@id"] = formId
+        }
+        if(Object.keys(record).length === 0){
+            //There is no good reason for this, but DEER allows it.  However, there better a type otherwise it is completely undescribed.
+            UTILS.warning("The form submitted does not contain any inputs.  The resulting object will not be described.", this.elem)
+        }
+        if(this.type) {record.type = this.type}
+        else{
```

Comment:
Oof this is wrong.  It should be more like If Object,keys().length === 0 && !this.type

Reply:
Done deal,